### PR TITLE
JavaScript Constructors Nathan Chon

### PR DESCRIPTION
**Pull request by Nathan-Chon**
animation42

### CODE REVIEW

File: javascript-constructors/example-constructor.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+function ExampleConstructor() {
+}
+
```
Comment (robgmerrill) [inline javascript-constructors/example-constructor.js:3]: per the instructions we should be using the `prototype` property. The `.getPrototypeOf()` method is going to get the prototype of `ExampleConstructor`. That would be the object that `ExampleConstructor` is inheriting from. Instead we want the `prototype` property which will be the object that it's children inherit from. 

eg: `ExampleConstructor.prototype`

File: javascript-constructors/example-constructor.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+console.log('value of prototype:', Object.getPrototypeOf(ExampleConstructor));
+console.log('typeof prototype:', typeof Object.getPrototypeOf(ExampleConstructor));
```
Comment (robgmerrill) [inline javascript-constructors/example-constructor.js:5]: Same comment as on line 4

File: javascript-constructors/example-constructor.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+var afunction = new ExampleConstructor();
+console.log('value of afunction', afunction);
+
```
Comment (robgmerrill) [inline javascript-constructors/example-constructor.js:9]: We also want to log the `typeof` `afunction`.

File: javascript-constructors/example-constructor.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+function ExampleConstructor() {
+}
+
+console.log('value of prototype:', Object.getPrototypeOf(ExampleConstructor));
+console.log('typeof prototype:', typeof Object.getPrototypeOf(ExampleConstructor));
+
+var afunction = new ExampleConstructor();
```
Comment (robgmerrill) [inline javascript-constructors/example-constructor.js:7]: Consider renaming. When we call a constructor function it creates an object. A variable name such as `anObject` might be a better indicator of the value it holds.

**Review by robgmerrill — CHANGES_REQUESTED** [review]
Close! A couple small changes requested!

**Comment by Nathan-Chon** [comment]
image
Revised version resubmitted

**Review by robgmerrill — APPROVED** [review]
👍

**Pull request merged**